ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,281 @@
+        CHECK(objects.size() == 0);
+
+        // No matching objects with listSingle. Object size should be 0.
+        CHECK(conn.list_objects(prefix + file_name, objects, list_single) == 0);
+        CHECK(objects.size() == 0);
+
+        // Put objects to prepare for test.
+        for (int i = 0; i < num_objects; i++) {
+            auto blob_client =
+              azure_client.GetBlockBlobClient(prefix + file_name + std::to_string(i) + ".txt");
+            blob_client.UploadFrom(file_name);
+        }
+
+        // List all objects. This is the size of num_objects.
+        CHECK(conn.list_objects(prefix + file_name, objects, !list_single) == 0);
+        CHECK(objects.size() == num_objects);
+
+        // List single. Object size should be 1.
+        objects.clear();
+        CHECK(conn.list_objects(prefix + file_name, objects, list_single) == 0);
+        CHECK(objects.size() == 1);
+
+        // There should be 2 matches with 'list_obj_1' prefix pattern.
+        objects.clear();
+        CHECK(conn.list_objects(prefix + file_name + "1", objects, !list_single) == 0);
+        CHECK(objects.size() == 2);
+
+        // Clean up.
+        for (int i = 0; i < num_objects; i++) {
+            auto blob_client =
+              azure_client.GetBlockBlobClient(prefix + file_name + std::to_string(i) + ".txt");
+            blob_client.Delete();
+        }
+    }
+
+    SECTION("(Testing delete functionality for Azure).", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "delete_obj_");
+
+        auto blob_client = azure_client.GetBlockBlobClient("delete_obj_" + file_name);
+
+        blob_client.UploadFrom(file_name);
+
+        // Test that an object can be deleted.
+        CHECK(conn.delete_object(file_name) == 0);
+
+        // Test that removing an object that doesn't exist returns a ENOENT.
+        CHECK(conn.delete_object(non_exi_object_key) == ENOENT);
+    }
+
+    SECTION("(Checking put functionality in Azure)", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "put_obj_");
+
+        CHECK(conn.put_object(file_name, path) == 0);
+
+        auto blob_client = azure_client.GetBlockBlobClient("put_obj_" + file_name);
+        blob_client.Delete();
+
+        // Test that putting an object that doesn't exist locally returns -1.
+        CHECK(conn.put_object(non_exi_object_key, non_exi_file_name) == -1);
+    }
+
+    SECTION("(Checking read functionality in Azure)", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "read_obj_");
+
+        auto blob_client = azure_client.GetBlockBlobClient("read_obj_" + file_name);
+
+        blob_client.UploadFrom(file_name);
+        void *buffer = calloc(1024, sizeof(char));
+
+        // Test reading whole file.
+        CHECK(conn.read_object(file_name, 0, payload.length(), buffer) == 0);
```

Comment:
I would add a new case where we read from the middle of the object.

Reply:
This test case is covered in the complex test case I can shift it up

Reply:
shifted up since complex test case has been removed